meta1v2/meta1_backend_services.c

```diff
@@ -1019,12 +1022,44 @@ meta1_backend_services_poll(struct meta1_backend_s *m1,
 	return err;
 }
 
+static GError *
+__services_list_append_account_and_container(gchar ***interm_result,
+		struct oio_url_s *url)
+{
+	GError *err = NULL;
+
+	const char *acc = oio_url_get(url, OIOURL_ACCOUNT);
+	if (acc == NULL) {
+		err = NEWERROR(CODE_INTERNAL_ERROR, "[%s] Unable to add account "
+				"to services list.", __FUNCTION__);
+		goto end;
+	}
+
+	gchar *packed_acc = g_strdup_printf(SQLX_ADMIN_ACCOUNT ":%s", acc);
+	*interm_result = oio_strv_append(*interm_result, packed_acc);
+
+	const char *cont = oio_url_get(url, OIOURL_USER);
+	if (cont == NULL) {
+		err = NEWERROR(CODE_INTERNAL_ERROR, "[%s] Unable to add reference name"
+				" to services list.", __FUNCTION__);
```

Comment:
IMO it's more cleaner to free `inter_result` here, in error code path, instead of doing it at l.1107